### PR TITLE
Add guide to canary deployment with Carvel Packages

### PR DESCRIPTION
**Pull request by izabelacg**
# Which other branches do you want a technical writer to cherry-pick this PR to (if any)?

Documentation for TAP 1.6

### CODE REVIEW

File: scc/canary-deployment.hbs.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+2. Add the label `app.kubernetes.io/name` to your `Workload`.
+   
+    The target deployment must have a single label selector in the format `app: <DEPLOYMENT-NAME>`.
+    In addition to `app`, Flagger supports `name` and `app.kubernetes.io/name` selectors.
+    
+    > **Note** You can use `tanzu apps workload apply <WORKLOAD-NAME> --label "app.kubernetes.io/name=<WORKLOAD-NAME>"` to modify an existing workload and add the required label.
+
+3. Add HTTPProxy to the deployment.
+
+4. Create a Canary resource that defines a canary release with progressive traffic shifting.
```
Comment (nickjameswebb) [inline scc/canary-deployment.hbs.md:40]: Can we include examples for these steps?

File: scc/canary-deployment.hbs.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+such as the success rate of HTTP requests, average request duration, and the health of the application's pods.
+By analyzing these key indicators, Flagger determines whether to promote the canary deployment to a wider audience or
+abort it if any issues arise.
+
+For creating canary releases, you can refer to the guide available at [Contour Canary Deployments](https://docs.flagger.app/tutorials/contour-progressive-delivery).
+In summary, the process explained in the guide above involves the following steps:
+
+1. Install Flagger on your Kubernetes cluster.
+
+2. Add the label `app.kubernetes.io/name` to your `Workload`.
```
Comment (nickjameswebb) [inline scc/canary-deployment.hbs.md:31]: Is the label automatically applied to the underlying K8s deployment? Just to confirm

**Comment by nickjameswebb** [comment]
We also need to update `toc.hbs.md` so that this page appears in the sidebar

File: scc/canary-deployment.hbs.md (hunk 2 above)
Comment (izabelacg) [reply scc/canary-deployment.hbs.md:31]: Yes, I confirmed the label gets applied to the k8s deployment 👍🏼

**Pull request merged**